features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/dashdirect/DashDirectViewModel.kt

```diff
@@ -207,49 +191,35 @@ class DashDirectViewModel @Inject constructor(
 
     suspend fun logout() = repository.logout()
 
-    fun saveGiftCard(
-        txId: Sha256Hash,
-        data: GetGiftCardResponse.Data
-    ) {
-        logOnPurchaseEvents()
-        val giftCardId = "${ServiceName.DashDirect.lowercase()}+${data.id ?: -1}" // e.g. dashdirect+1234
+    fun saveGiftCardDummy(txId: Sha256Hash, orderId: String, paymentId: String) {
         val giftCard = GiftCard(
-            id = giftCardId,
+            id = UUID.randomUUID().toString(),
```

Comment:
Because gift card ID isn't available at the time we save the dummy record, I figure it's better to have a UUID here and separate fields for the service.